packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -285,108 +421,112 @@ export default class API {
 
       if (!this.useOpenAuthoring) {
         return this.request(`${this.repoURL}/contents/${key}.json`, metadataRequestOptions)
-          .then(response => JSON.parse(response))
+          .then((response: string) => JSON.parse(response))
           .catch(errorHandler);
       }
 
       const [user, repo] = key.split('/');
       return this.request(`/repos/${user}/${repo}/contents/${key}.json`, metadataRequestOptions)
-        .then(response => JSON.parse(response))
+        .then((response: string) => JSON.parse(response))
         .catch(errorHandler);
     });
   }
 
-  retrieveContent(path, branch, repoURL) {
+  retrieveContent({ path, branch, repoURL, parseText }: ContentArgs) {
     return this.request(`${repoURL}/contents/${path}`, {
-      headers: { Accept: 'application/vnd.github.VERSION.raw' },
       params: { ref: branch },
       cache: 'no-store',
-    }).catch(error => {
-      if (hasIn(error, 'message.errors') && find(error.message.errors, { code: 'too_large' })) {
-        const dir = path
-          .split('/')
-          .slice(0, -1)
-          .join('/');
-        return this.listFiles(dir, { repoURL, branch })
-          .then(files => files.find(file => file.path === path))
-          .then(file => this.getBlob(file.sha, { repoURL }));
-      }
-      throw error;
-    });
-  }
-
-  readFile(path, sha, { branch = this.branch, repoURL = this.repoURL } = {}) {
+    }).then((file: GitHubFile) => this.getBlob({ sha: file.sha, repoURL, parseText }));
+  }
+
+  readFile(
+    path: string,
+    sha: string | null,
+    {
+      branch = this.branch,
+      repoURL = this.repoURL,
+      parseText = true,
+    }: {
+      branch?: string;
+      repoURL?: string;
+      parseText?: boolean;
+    } = {},
+  ) {
     if (sha) {
-      return this.getBlob(sha);
+      return this.getBlob({ sha, repoURL, parseText });
     } else {
-      return this.retrieveContent(path, branch, repoURL);
+      return this.retrieveContent({ path, branch, repoURL, parseText });
     }
   }
 
-  fetchBlob(sha, repoURL) {
-    return this.request(
-      `${repoURL}/git/blobs/${sha}`,
-      {
-        headers: { Accept: 'application/vnd.github.VERSION.raw' },
-      },
-      response => response,
-    );
-  }
-
-  async fetchBlobContent(sha, repoURL) {
-    const response = await this.fetchBlob(sha, repoURL);
-    const text = await response.text();
+  async fetchBlobContent({ sha, repoURL, parseText }: BlobArgs) {
```

Comment:
Using `Accept: 'application/vnd.github.VERSION.raw'` always returns `text/plain` as content type, which once again trips our recording proxy.
Regardless it is odd that images are returned as a string and not blog...